### ISSUE TITLE
[Discussion] who is using `nestia` in production?

### ISSUE THREAD

**Issue by samchon** [samchon/nestia#203]
If you're using `nestia` in production level, can you introduce me?

It would better if you are pasting production homepage URL with logo image.

**Comment by dipanshuhappy** [comment]
I am using nestia in prod
here is the repo https://github.com/viaprize/viaprize

**Comment by dipanshuhappy** [comment]
> If you're using `nestia` in production level, can you introduce me?
> 
> It would better if you are pasting production homepage URL with logo image.

what do you think of swagger code gen

**Comment by samchon** [comment]
The swagger code gen means something like this?

https://nestia.io/docs/migrate/

**Comment by Valetek** [comment]
I'm currently testing it in production, I have no url to provide as it's all in private

But for what I could see Nestia is really helpful for what the controller decorator provide, such as data type transformation/validation

I'm still not sure about the sdk for swagger as currently there is some lib that create DTO with Swagger decorator all extracted from your db schema (in my POV that seem as a more secure way of keeping the docs updated as there is no human manipulation) and the sdk look painful to maintain for docs

Except for the swagger part that is personnal, when configured nestia can't be push to production without manual modification

In production we don't install the package in devDendencies, and you have to run the ```prepare``` command to make nestia work on service start, if ```ts-patch``` and ```typescript``` are not moved to dependencies, your service will not be able to start in production

in my POV ```typescript``` should not be in production, so maybe there is something to improve here ? 

If you have some information about any of the point I wrote, I would be happy to discuss about it, maybe it's only me that missed some step in your documentation

**Comment by bogeeee** [comment]
Yes, splitting it into transformer and runtime packages would be nice. 
@Valetek: You can run ts-patch in life mode (without patching). See [here](https://github.com/bogeeee/restfuncs?tab=readme-ov-file#setting-up-the-build-here-it-gets-a-bit-nasty-).

**Comment by bogeeee** [comment]
I'm looking for a swagger code gen too. But in the other direction: Generating swagger spec from your typescript types. 
Not sure if it's Typia's purpose to solve this, or this might be a new project...
But it would be cool to have such in a ts-patch transformer, so i could integrate that into the restfuncs ecosystem and give the user a zero-conf experience for downloading openAPI spec.

**Comment by Valetek** [comment]
Some tools provide swagger generating tools, as prisma community do

**Comment by machendos** [comment]
@samchon we with big team are rewriting the whole Backend using Nestia. Code is not an opensource thugh. We're facing some limitations and bugs, but successfully found solutions for most of them. So we decided to continue

**Comment by samchon** [comment]
> @samchon we with big team are rewriting the whole Backend using Nestia. Code is not an opensource thugh. We're facing some limitations and bugs, but successfully found solutions for most of them. So we decided to continue

If solved some limitations and bugs, please report those bugs or share some knowhows please.

**Comment by nezudevv** [comment]
I am looking into using this in Production as well, however, the only issue I am running into is the requirement for explicit return types on Controller Methods. Due to us using a query builder that automatically handles return type inference for us we avoid having to manually create response types for each controller. For me to get the team on board it would be easier if the explicit return type wasn't required.

**Comment by samchon** [comment]
> I am looking into using this in Production as well, however, the only issue I am running into is the requirement for explicit return types on Controller Methods when generating swagger docs. Due to us using a query builder that automatically handles return type inference for us we avoid having to manually create response types for each controller. For me to get the team on board it would be easier if the explicit return type wasn't required.

Upgrade to v3.9.0 patch, then `npx nestia swagger` will not check implicit type.

However, `npx nestia sdk` still validates it.

**Comment by nezudevv** [comment]
> > I am looking into using this in Production as well, however, the only issue I am running into is the requirement for explicit return types on Controller Methods when generating swagger docs. Due to us using a query builder that automatically handles return type inference for us we avoid having to manually create response types for each controller. For me to get the team on board it would be easier if the explicit return type wasn't required.
> 
> Upgrade to v3.9.0 patch, then `npx nestia swagger` will not check implicit type.
> 
> However, `npx nestia sdk` still validates it.

Thank you. Is there currently support for allowing the swagger docs to generate the response type based on type inference, without having to specify the type? I believe the only library i've seen do this is TSOA but it'd be amazing if Nestia had this as well.

**Comment by Rafik-Belkadi-malou** [comment]
We are using Nestia in production for our startup: 

[Recap app](https://mail.recap-app.com)

it's a mail client on top of gmail
